hal/armv7m/stm32/l4/console.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Phoenix-RTOS
+ *
+ * plo - operating system loader
+ *
+ * Console
+ *
+ * Copyright 2021 Phoenix Systems
+ * Authors: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include "stm32l4.h"
+
+
+static struct {
+	volatile u32 *base;
+	unsigned cpufreq;
+} halconsole_common;
+
+
+enum { cr1 = 0, cr2, cr3, brr, gtpr, rtor, rqr, isr, icr, rdr, tdr };
+
+
+void hal_consolePrint(const char *s)
+{
+	while (*s) {
+		if (~(*(halconsole_common.base + isr)) & 0x80)
+			continue;
+
+		*(halconsole_common.base + tdr) = *(s++);
+	}
+
+	while (~(*(halconsole_common.base + isr)) & 0x80);
+
+	return;
+}
+
+
+void console_init(void)
+{
+	struct {
+		void *base;
```

Comment:
Lack of `volatile`

Reply:
It's not needed, as You pointed out before